### ISSUE TITLE
gpu: Our 3D coordinate system is not standard (although nothing is)

### ISSUE THREAD

**Issue by rcoreilly** [cogentcore/core#1121]
### Describe the bug

The whole tortured history of different coordinate systems in different 3D frameworks is crazy: https://github.com/gpuweb/gpuweb/issues/416   Because we went through OpenGL to vulkan in vgpu (which is a real outlier -- nice job folks!), we have a key inconsistency with what most people will probably do starting fresh with WebGPU.  And it is internally inconsistent with the shapes mesh in some subtle way.

For example, https://github.com/rajveermalviya/go-webgpu-examples has everything using FrontFaceCCW, but we use pl.SetFrontFace(wgpu.FrontFaceCW) in gpipeline defaults.

Here's the actual WebGPU coordinate system with a diagram: https://www.w3.org/TR/webgpu/#coordinate-systems , which looks like this in ascii:

```
    ^
 Y+ | 
    |
    +-------->
   /      X+
  / Z-
 v
```

This is a *left-handed* coordinate system, shared with DirectX and Metal.

Vulkan is *right-handed*, but with Y- being "up", which makes no sense at all!:

```
    ^
 Y- | 
    |
    +-------->
   /      X+
  / Z-
 v
```

Therefore, we had systematically inverted that using a different "VkProjection" matrix, so that Y was effectively up again.

However, inconsistent with all of these coordinate systems, our standard "world" coordinates are based on a *right-handed* coordinate system that has Z+ coming toward you in the depth plane.

```
    ^
 Y+ | 
    |
    +-------->
   /      X+
  / Z+
 v
```

There is some confusion in the different references in my searching, but this also seems to be the standard OpenGL coordinate system, eg: https://www3.ntu.edu.sg/home/ehchua/programming/opengl/CG_BasicsTheory.html

This coordinate system is consistent with the broader world outside of 3D rendering, where right-handed coordinates are preferred in physics and engineering: https://en.wikipedia.org/wiki/Right-hand_rule, and in many graphical frameworks, the general world space is right-handed: https://www.kdab.com/projection-matrices-with-vulkan-part-1/ (this has OpenGL going the other way though).  

In the "standard cartesian coordinate system" (https://en.wikipedia.org/wiki/Cartesian_coordinate_system), everything is rotated 90 degrees along the X axis, so that Y+ now points into the depth plane, and Z+ points upward, so in effect you have vertical "stacks" of standard X-Y coordinates:

```
    ^   ^
 Z+ |  / Y+
    | / 
    +-------->
   /      X+
  / Y-
 v
```

Critically, this is consistent with our preferred right-hand coordinate system, just with a rotation -- if you instead "swap" the Y, Z axes without rotating, as happens in WebGPU et al, then you end up with a left-hand coordinate system.

Here's a summary of some other "standards" in different 3D frameworks:

https://x.com/FreyaHolmer/status/644881436982575104 
<img width="568" alt="image" src="https://github.com/user-attachments/assets/f70e7d50-79dd-4442-a765-b7a19d9199dd">

Basically, there are more right-hand than left hand overall.

So, after all that, we should stick with the existing coordinate system and just accept that it will be inconsistent with other WebGPU tutorials etc.  Will add this stuff to the docs to make it clear.

And should fix the Plane NormNeg name so that is clearer.

### How to reproduce

after all this, not actually a bug.

### Platform

macOS

**Comment by kkoreilly** [comment]
Right-handed, Z is up makes the most logical sense though, right? Shouldn't we do that instead of our existing system? This is the classic 3-space in math:

![3-Space](https://upload.wikimedia.org/wikipedia/commons/thumb/8/83/Coord_planes_color.svg/1024px-Coord_planes_color.svg.png)

Which is this in ASCII, right?

```
    ^   ^
 Z+ |  / X-
    | / 
    +-------->
   /      Y+
  / X+
 v
```

**Comment by rcoreilly** [comment]
That is an absolutely _terrible_ system!  First of all, it is just a 90 degree rotation around the Z axis of the other standard cartesian representation, so it is entirely compatible in terms of right-handedness etc.

But it has the Y axis being the "horizontal" axis which runs entirely counter to the standard 2D cartesian system!  Absolutely unforgivable.  And X is now the depth axis?  no way!

Furthermore, I didn't say this earlier but the key problem with the other cartesian one shown above with Y+ going in depth, is that the 2D planar screen should be XY, for the same reason -- the anchor in everything is the standard 2D system, which as far as I know is actually entirely standard and conventional.

**Comment by kkoreilly** [comment]
It is the absolute standard 3D coordinate system in math (see https://en.wikipedia.org/wiki/Three-dimensional_space). I am absolutely fine with this though:

```
    ^   ^
 Z+ |  / Y+
    | / 
    +-------->
   /      X+
  / Y-
 v
```

**Comment by kkoreilly** [comment]
In terms of having a rational 3D extrapolation of a standard 2D plane, that system I put in ASCII above (and you put earlier), is by far the best.

**Comment by kkoreilly** [comment]
The 3-space one I sent before is very handy (in my experience) when doing hand drawings of 3D things in math for various reasons, but it is just a rotated perspective on the standard cartesian representation in ASCII below, which probably makes the most logical sense for computers. Just to be clear, this is my proposal:

```
    ^   ^
 Z+ |  / Y+
    | / 
    +-------->
   /      X+
  / Y-
 v
```

**Comment by kkoreilly** [comment]
I guess we can keep the existing right-handed Y-up system.
